lib/logstash/outputs/tcp.rb

```diff
@@ -204,7 +176,38 @@ def server?
   end # def server?
 
   public
-  def receive(event)
-    @codec.encode(event)
-  end # def receive
+  def multi_receive_encoded(encoded)
+    if server?
+      @server_mutex.synchronize do
+        @client_threads.each do |client_thread|
+           encoded.each do |event,data|
+             client_thread[:client].write(data)
+           end
+        end
+        @client_threads.reject! {|t| !t.alive? }
```

Comment:
What is the purpose of this? Why would some client threads be alive? 

I know this is original code but it's confusing to me. I don't understand it.